app/src/main/java/com/infomaniak/mail/ui/main/onboarding/PermissionsOnboardingPagerFragment.kt

```diff
@@ -0,0 +1,106 @@
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.main.onboarding
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.viewpager2.widget.ViewPager2
+import com.infomaniak.lib.core.utils.context
+import com.infomaniak.lib.core.utils.safeBinding
+import com.infomaniak.lib.core.utils.safeNavigate
+import com.infomaniak.mail.R
+import com.infomaniak.mail.data.LocalSettings
+import com.infomaniak.mail.databinding.FragmentPermissionsOnboardingPagerBinding
+import com.infomaniak.mail.ui.MainViewModel
+import com.infomaniak.mail.utils.PermissionUtils
+import com.infomaniak.mail.utils.removeOverScrollForApiBelow31
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class PermissionsOnboardingPagerFragment : Fragment() {
+
+    private var binding: FragmentPermissionsOnboardingPagerBinding by safeBinding()
+    private val mainViewModel: MainViewModel by activityViewModels()
+    private var currentPosition = 0
+
+    @Inject
+    lateinit var localSettings: LocalSettings
+
+    @Inject
+    lateinit var permissionUtils: PermissionUtils
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return FragmentPermissionsOnboardingPagerBinding.inflate(inflater, container, false).also { binding = it }.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?): Unit = with(binding) {
+        super.onViewCreated(view, savedInstanceState)
+
+        permissionUtils.apply {
+            registerReadContactsPermission(fragment = this@PermissionsOnboardingPagerFragment)
+            registerNotificationsPermissionIfNeeded(fragment = this@PermissionsOnboardingPagerFragment)
+        }
+
+        permissionsViewpager.apply {
+            adapter = PermissionsPagerAdapter(childFragmentManager, viewLifecycleOwner.lifecycle)
+            isUserInputEnabled = false
+            savedInstanceState?.getInt(VIEW_PAGER_POSITION_KEY)?.let { setCurrentItem(it, false) }
+            removeOverScrollForApiBelow31()
+        }
+
+        continueButton.setOnClickListener {
```

Comment:
Clicking on "next" on API 25 did not prompt me to to give contacts permissions when I updated, it directly skipped to the ThreadListFragment. Maybe I had manually taken the rights away or I had already chosen to not given them back then so the system did not want to prompt me again?

Reply:
I try and didn't have any bug, I think you probably ticked the "don't ask me" box last one you used that emu